libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxOffboardSpeechPlayer.kt

```diff
@@ -0,0 +1,120 @@
+package com.mapbox.navigation.ui.voice.api
+
+import android.content.Context
+import android.media.MediaPlayer
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayer
+import com.mapbox.navigation.ui.base.api.voice.SpeechPlayerCallback
+import com.mapbox.navigation.ui.base.model.voice.Announcement
+import com.mapbox.navigation.ui.base.model.voice.SpeechState
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileNotFoundException
+import java.io.IOException
+
+/**
+ * Online implementation of [SpeechPlayer].
+ * Will retrieve synthesized speech mp3s from Mapbox's API Voice.
+ * @property context Context
+ * @property accessToken String
+ * @property language String
+ */
+internal class MapboxOffboardSpeechPlayer(
+    private val context: Context,
+    private val accessToken: String,
+    private val language: String
+) : SpeechPlayer {
+
+    private var mediaPlayer: MediaPlayer? = null
+    private var volumeLevel: Float = DEFAULT_VOLUME_LEVEL
+    private lateinit var clientCallback: SpeechPlayerCallback
+    private var currentPlay: SpeechState.Play = SpeechState.Play(Announcement("", null, null))
```

Comment:
```suggestion
    private var currentPlay: SpeechState.Play? = null
```

up to you but i prefer `null`, that way the compiler will help make sure empty announcements are properly handled

Reply:
The idea here was that `currentPlay` couldn't be `null` 😅

Reply:
what happens when it's empty? do we need to filter them? does the caller need to filter them?

Reply:
https://github.com/mapbox/mapbox-navigation-android/blob/9d9590ccafcb89459af455a9fe70b84435ce81d4/libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxOffboardSpeechPlayer.kt#L43-L44

Reply:
> that way the compiler will help make sure empty announcements are properly handled

I like that as well :+1: